### PR TITLE
[Proposal] Update CONTRIBUTING and issue template(s) for feature requests

### PR DESCRIPTION
**Pull request by Airblader**
Once we agreed on the approach for this, there are further points such as

- [x] Update this in the project goals on the website's index page? (https://github.com/i3/i3.github.io/pull/101)
- [x] Extend i3bot to be a bit smarter about replies for this (tbd by @stapelberg)
- [x] I'd still like to make this a bit more "public" by publishing this move on, say, reddit. (tbd by @Airblader)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -28,6 +28,15 @@ Describe the desired behavior you expect after mitigation of the issue,
 e.g., »The window left next to the current window should be focused.«
 -->
 
+## Impact
+<!--
+Please note that at this point we focus on maintaining i3 and fixing bugs, and will rarely consider features which require further configuration or significant complexity.
+In such cases you should consider and present specific benefits derived from adding this feature such that it can be weighed against the cost of additional complexity and maintenance.
+-->
+<pre>
+[ ] This feature requires new configuration and/or commands
```
Comment (orestisfl) [inline .github/ISSUE_TEMPLATE/feature_request.md:37]: We can auto-label this with i3bot as well.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -18,6 +18,18 @@ Note that bug reports and feature requests for related projects should be filed
...
 
+## i3 feature requests
+1. Read the [project goals](https://i3wm.org) on the website and make sure that
+   they are compatible with the feature you want to suggest.
+2. We are generally happy with the current feature set of i3 and instead focus
+   on maintenance such as stability and fixing bugs. New features will rarely
+   be considered if they require additional configuration and/or commands, or
+   if they add significant complexity (either through the exposed configuration
+   or mental complexity) to the project.
+3. Explain in detail what problem the feature addresses.
```
Comment (stapelberg) [inline .github/CONTRIBUTING.md]: Maybe add “and why existing features fall short”

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -18,6 +18,18 @@ Note that bug reports and feature requests for related projects should be filed
    encountered the issue you are about to report while using a compositor,
    please try reproducing it without a compositor.
 
+## i3 feature requests
```
Comment (stapelberg) [inline .github/CONTRIBUTING.md:21]: nit: add blank line after heading

**Review by stapelberg — APPROVED** [review]

**Comment by Airblader** [comment]
@stapelberg Would you be able to take care of i3bot (afterwards)? Go isn't exactly my language… :-)

**Comment by stapelberg** [comment]
> @stapelberg Would you be able to take care of i3bot (afterwards)? Go isn't exactly my language… :-)

Yes, just let me know what needs to be changed precisely.

**Review by stapelberg — APPROVED** [review]

**Comment by Airblader** [comment]
I would propose something like whenever the issue is determined to be a feature request, reply with something like

> Please note that new features which require additional configuration will usually not be considered. We are happy with the feature set of i3 and want to focus in fixing bugs instead. We do accept feature requests, however, and will evaluate whether the added benefit (clearly) outweighs the complexity it adds to i3.

**Comment by Airblader** [comment]
… and of course do what @orestisfl suggested, i.e. tag the issue with some (new) tag like "requires-configuration" when the checkbox is detected and detected to be checked.

**Comment by Airblader** [comment]
I've also posted about this on reddit now: https://www.reddit.com/r/i3wm/comments/kn8pa2/an_update_on_the_future_of_i3/

**Comment by Airblader** [comment]
Also to avoid some confusion I've removed the accepted label from all feature requests and replaced it with tentatively_accepted.

**Pull request merged**